wallet/src/main/scala/org/constellation/wallet/Transaction.scala

```diff
@@ -63,7 +63,7 @@ object TransactionExt {
   ): Transaction = {
     val lastTxRef =
       prevTx
-        .map(tx => LastTransactionRef(tx.hash, tx.lastTxRef.ordinal + 1))
+        .map(tx => LastTransactionRef(tx.hash, tx.ordinal))
```

Comment:
Why?

Reply:
The main reason is easier mocking.I don't need to mock subobjects, I can do it with just `tx.ordinal should return 123`. The secondary reason is DRY.

Reply:
But it's a place where it's defined how to create transaction. It should stay as it is and we shouldn't rely on the implementation part (so the additional method in the tx)

Reply:
Well in the same line we do rely on the implementation by taking: `tx.hash` instead of `tx.edge.observationEdge.hash`.

Reply:
That's true but we rely on the hash method in the whole codebase in the same way. We assume that there is a `hash` for every class which is signable. For the ordinal I see a reason why it's explicitly defined there but it's up to you :)

Reply:
I still can't understand why would you advocate for duplicating the method implementation, given the argument:
>it's a place where it's defined how to create transaction

Can other guys give their opinion about this issue @kpudlik  @TheMMaciek? Maybe I'm missing something.

Reply:
I get both points, from one side the definition of ordinal should be set in stone so we should be able to reuse it, from to other relying on something that should be set in stone but is defined somewhere outside of current scope and relying on it feels uncertain. After all I think we are aware enough that it's crucial to not change transaction's logic freely and our team is small so we should catch it if someone fiddles with it. Therefore it's ok for me to reuse ordinal definition.

Reply:
@tbekas - I agree with @TheMMaciek

Reply:
So to close the discussion, @TheMMaciek you are in favor of keeping the change, am I reading it right?

Reply:
Yes :).

Reply:
Touching that part is risky but I get all your points and it is fine for me to keep that change :)